nomt/src/bitbox/mod.rs

```diff
@@ -158,85 +114,39 @@ impl DB {
                         changed_meta_pages.insert(meta_map.page_index(bucket as usize));
                     }
 
-                    // fill the WalBatch and the pages that need to be written to disk
-                    wal_batch.append_entry(WalEntry::Update {
-                        page_id: page_id.encode(),
-                        changed: get_changed(&page, &page_diff),
-                        page_diff: page_diff.get_raw(),
-                        bucket_index: bucket as u64,
-                    });
+                    let changed = get_changed(&page, &page_diff);
+                    wal_blob_builder.write_update(
+                        page_id.encode(),
+                        page_diff.get_raw(),
+                        changed,
```

Comment:
noting an optimization possibility here (for a later PR)

the impact of memory allocation in `get_changed` is not insignificant. altering `wal_blob_builder.write_update` to accept an iterator as an argument would avoid that.